packages/terra-date-picker/src/react-datepicker/index.jsx

```diff
@@ -51,84 +51,297 @@ const WrappedCalendar = onClickOutside(Calendar)
       PropTypes.string,
       PropTypes.array
     ]),
+    /**
+     * Date format displayed on calendar header.
+     */
     dateFormatCalendar: PropTypes.string,
+    /**
+     * A callback function for custom day class names.
+     */
     dayClassName: PropTypes.func,
+    /**
+     * Prop to disable dates on calendar.
+     */
     disabled: PropTypes.bool,
+    /**
+     * Prop to disable keyboard navigation.
+     */
     disabledKeyboardNavigation: PropTypes.bool,
+    /**
+     * Prop to select a dropdown mode for the year dropdown.
+     */
     dropdownMode: PropTypes.oneOf(['scroll', 'select']).isRequired,
+    /**
+     * Maximum Date value for a date range.
+     */
     endDate: PropTypes.object,
+    /**
+     * Array to store values of date that are disabled to pick .
+     */
     excludeDates: PropTypes.array,
+    /**
+     * Filter specific dates that can be selected only .
+     */
     filterDate: PropTypes.func,
+    /**
+     * Specifies whether the height of calendar dom fixed or variable.
+     */
     fixedHeight: PropTypes.bool,
+    /**
+     * A callback function to format week number .
+     */
     formatWeekNumber: PropTypes.func,
+    /**
+     * Highlight range of dates with custom classes.
+     */
     highlightDates: PropTypes.array,
+    /**
+     * Value of custom input id.
+     */
     id: PropTypes.string,
+    /**
+     * Show dates only in the given array.
+     */
     includeDates: PropTypes.array,
+    /**
+     * Prop to render Inline version of datepicker component.
+     */
     inline: PropTypes.bool,
     /**
      * @private
      * Internationalization object with translation APIs. Provided by `injectIntl`.
      */
     intl: intlShape,
+    /**
+     * Prop to determine whether or not the date picker is clearable.
+     */
     isClearable: PropTypes.bool,
+    /**
+     * Name of locale data for different international formatting.
+     */
     locale: PropTypes.string,
+    /**
+     * Maximum Value of date that can be selected by user.
+     */
     maxDate: PropTypes.object,
+    /**
+     * Minimum Value of date that can be selected by user.
+     */
     minDate: PropTypes.object,
+    /**
+     * Prop to show multiple months on date picker.
+     */
     monthsShown: PropTypes.number,
+    /**
+     * Value for name of custom input.
+     */
     name: PropTypes.string,
+    /**
+     * A callback function to execute when object loses focus.
+     */
     onBlur: PropTypes.func,
+    /**
+     * A callback function to execute when date is selected.
+     */
     onChange: PropTypes.func.isRequired,
+    /**
+     * A callback function to execute when value is selected from dropdown.
+     */
     onSelect: PropTypes.func,
+    /**
+     * A callback function to execute when week is selected.
+     */
     onWeekSelect: PropTypes.func,
+    /**
+     * A callback function to execute when nothing is selected.
```

Comment:
Can be changed to "A callback function to execute when outside the date picker is clicked."